text/0401-coercions.md

```diff
@@ -319,20 +319,23 @@ descriptions are equivalent.
 Casting is indicated by the `as` keyword. A cast `e as U` is valid if one of the
 following holds:
 
-* `e` has type `T` and `T` coerces to `U`;
-
-* `e` has type `*T` and `U` is `*U_0` (i.e., between any raw pointers);
-
-* `e` has type `*T` and `U` is `uint` , or vice versa;
-
-* `e` has type `T` and `T` and `U` are any numeric types;
-
-* `e` is a C-like enum and `U` is any integer type, `bool`;
-
-* `e` has type `T` and `T == u8` and `U == char`;
-
-* `e` has type `T` and `T == &[V, ..n]` or `T == &V` and `U == *const V`, and
-  similarly for the mutable variants to either `*const V` or `*mut V`.
+ * `e` has type `T` and `T` coerces to `U`; *coercion-cast*
+ * `e` has type `*T`, `U` is `*U_0`, and either `U_0: Sized` or
+    unsize_kind(`T`) = unsize_kind(`U_0`); *ptr-ptr-cast*
+ * `e` has type `*T` and `U` is a numeric type, while `T: Sized`; *ptr-addr-cast*
+ * `e` is an integer and `U` is `*U_0`, while `U_0: Sized`; *addr-ptr-cast*
+ * `e` has type `T` and `T` and `U` are any numeric types; *numeric-cast*
+ * `e` is a C-like enum and `U` is an integer type; *enum-cast*
+ * `e` has type `bool` or `char` and `U` is an integer; *prim-int-cast*
+ * `e` has type `u8` and `U` is `char`; *u8-char-cast*
+ * `e` has type `&[T; n]` and `U` is `*const T`; *array-ptr-cast*
```

Comment:
Is there any reason for the mut version not to be here? Feels to me it ought to be for the sake of completeness, unless there is a good reason not to.

Is it worth specifying what happens if n == 0 for this kind of cast? I assume the cast succeeds, but what is the value of the pointer? (null? undefined? a valid pointer to something?)